README_python.md

```diff
@@ -170,68 +201,294 @@ essentially zero cost, with storage being shared. Thus, we can usually pass it
 directly to existing Python APIs (like `imsave()`) that expect 'image-like'
 objects without any explicit conversion necessary.
 
-## Using Halide Generators from Python
+## Halide Generators In Python
+
+In Halide, a "Generator" is a unit of encapsulation for Halide code. It is
+self-contained piece of code that can:
+-  Produce a chunk of Halide IR (in the form of an `hl.Pipeline`) that is appropriate for compilation (via either JIT or
+AOT)
+-  Expose itself to the build system in a discoverable way
+-  Fully describe
+itself for the build system with metadata for (at least) the type and number of
+inputs and outputs expected
+-  Allow for build-time customization of
+coder-specified parameters in a way that doesn't require editing of source code
+
+Originally, Halide only supported writing Generators in C++. In this document,
+we'll use the term "C++ Generator" to mean "Generator written in C++ using the
+classic API", the term "Python Generator" to mean "Generator written in Halide's
+Python bindings", and just plain "Generator" when the discussion is relatively
+neutral with respect to the implementation language/API.
+
+### Writing a Generator in Python
+
+A Python Generator is a class that: - has the `@hl.generator` decorator applied
```

Comment:
Consider using a bullet list here for readability.

Reply:
Ouch, it was one once -- the Markdown formatter scrambled it, fixing